### PR TITLE
Add marketo newsletter sign up form

### LINKED ISSUE #2391: add email link to Insights - build

### PR DESCRIPTION
**Pull request by grahambancroft**
## Done

Add marketo newsletter sign up form

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: [http://0.0.0.0:8023/](http://0.0.0.0:8023/)
- Navigate to an article, note that the 'interested in' category will be already selected based on article tags Fill out the form, you should be redirected to the same url at insights.ubuntu.com

## Issue / Card

Fixes https://github.com/canonical-websites/www.ubuntu.com/issues/2391

### CODE REVIEW

File: templates/post.html [hunk 1]
```diff
@@ -85,6 +85,81 @@
...
+        }
+        </script>
+
+        <form class='p-form mktoForm mktoNoJS' action='https://pages.canonical.com/index.php/leadCapture/save' method='post'>
+          <h2>Sign up for email updates</h2>
+          <p>Choose the topics you're interested in</p>
+            <ul class="p-list">
+              <li class="p-list__item">
+              <div>
+                <input class="mktoLogicalField mktoCheckboxList mktoHasWidth" type="checkbox" id="insightscloudserver" name="insightscloudserver" value="yes" {% if post.topics[0].slug == 'cloud' or 'server'  %} checked="checked"{% endif %} />
```
Comment (anthonydillon) [inline templates/post.html]: The `value` should be unique to the checkbox?

File: templates/post.html [hunk 2]
```diff
@@ -85,6 +85,81 @@
...
+            </ul>
+
+          <div class='mktoFormCol'>
+            <label class="off-left mktoLabel" for='Email'>Your email</label>
+            <input type="text" placeholder="Your email" class='mktoField mktoTextField' name='Email' id='Email' required>
+          </div>
+
+          <div class="mktField mktLblRight">
+            <span class="mktInput mktLblRight">
+              <input class="mktFormCheckbox" name="canonicalUpdatesOptIn" id="canonicalUpdatesOptIn" value="yes" tabindex="9" type="checkbox" />
```
Comment (anthonydillon) [inline templates/post.html]: Should have a unique value

File: templates/post.html [hunk 3]
```diff
@@ -85,6 +85,81 @@
...
+              <label for="" class="p-form__label">I would like to receive occasional updates from Canonical by email.</label>&nbsp;<span class="mktFormMsg"></span>
```
Comment (anthonydillon) [inline templates/post.html]: `for="canonicalUpdatesOptIn"`

**Review by anthonydillon — CHANGES_REQUESTED** [review]
A few code comments

File: templates/post.html (hunk 1 above)
Comment (grahambancroft) [reply templates/post.html]: Done

File: templates/post.html (hunk 2 above)
Comment (grahambancroft) [reply templates/post.html]: Done

File: templates/post.html (hunk 3 above)
Comment (grahambancroft) [reply templates/post.html]: Done

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Few QA comments

File: templates/post.html [hunk 4]
```diff
@@ -85,6 +85,81 @@ <h3>
...
+              </li>
+              <li class="p-list__item">
+                  <input class="mktoLogicalField mktoCheckboxList mktoHasWidth" id="insightstutorials" name="insightstutorials" value="Tutorials" {% if post.topics[0].slug == 'tutorials' %} checked="checked" {% endif %}type="checkbox">
+                  <label for="insightstutorials">Tutorials</label>
+              </li>
+            </ul>
+
+          <div class='mktoFormCol'>
+            <label class="off-left mktoLabel" for='Email'>Your email</label>
+            <input type="text" placeholder="Your email" class='mktoField mktoTextField' name='Email' id='Email' required>
```
[the remark this answers is not in the document]
Comment (grahambancroft) [reply templates/post.html]: Done

File: templates/post.html [hunk 5]
```diff
@@ -85,6 +85,81 @@ <h3>
         </p>
       </div>
     </div>
+
+    <div class="row">
+      <div class="col-8">
```
[the remark this answers is not in the document]
Comment (grahambancroft) [reply templates/post.html]: Done

File: templates/post.html [hunk 6]
```diff
@@ -85,6 +85,81 @@ <h3>
...
+          </div>
+
+          <div>
+            <span style="display:none;"><input type="text" name="_marketo_comments" value=""></span>
+            <span class='mktoButtonWrap'><button type='submit' class='mktoButton'>Subscribe now</button></span>
+            <input value="1959" class="mktoField mktoFieldDescriptor" name="lpId" type="hidden">
+            <input value="30" class="mktoField mktoFieldDescriptor" name="subId" type="hidden">
+            <input type="hidden" name="lpurl" value="https://pages.canonical.com/Insights-Subscription_Insights-Subscription-test.html?cr={creative}&amp;kw={keyword}" />
+            <input value="1212" class="mktoField mktoFieldDescriptor" name="formid" type="hidden">
+            <input type="hidden" name="ret" value="http://insights.ubuntu.com{{request.path}}?digest=true" />
```
[the remark this answers is not in the document]
Comment (grahambancroft) [reply templates/post.html]: Done, but it's not working, the return url in the code is good, but when you submit the form it redirects to ubuntu.com

**Comment by grahambancroft** [comment]
Ready for review again. The form now returns to the original article and displays a success message.

File: templates/post.html [hunk 7]
```diff
@@ -1,6 +1,18 @@
 {% extends "layout.html" %}
 
 {% block body %}
+{% if request.args.get('newsletter') == 'true' %}
```
Comment (anthonydillon) [inline templates/post.html]: The returned arg is `digest`

File: templates/post.html [hunk 8]
```diff
@@ -1,6 +1,18 @@
...
+<div class="p-strip--light  is-shallow">
+  <div class="row">
+    <div class="p-notification--positive">
+      <p class="p-notification__response">
+        <span class="p-notification__status">Success:</span>Thank you for subscribing! You will begin receiving emails as new content is posted. You may unsubscribe any time by clicking the link in the email.
```
Comment (anthonydillon) [inline templates/post.html]: Please add a space after the span

File: templates/post.html [hunk 9]
```diff
@@ -87,6 +99,72 @@ <h3>
...
 </div>
+<div class="p-strip is-shallow">
+    <div class="row">
+      <div class="col-8">
+        <form action="https://pages.canonical.com/index.php/leadCapture/save" method="post" id="mktoForm_1212" novalidate="novalidate">
+          <h2>Sign up for email updates</h2>
+          <p>Choose the topics you're interested in</p>
+            <ul class="p-list">
+              <li class="p-list__item">
+              <div>
```
Comment (anthonydillon) [inline templates/post.html]: Why does this one have a `div`?

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Few comments in the code. QA: the notification didn't appear when the form returned.

File: templates/post.html (hunk 9 above)
Comment (grahambancroft) [reply templates/post.html]: Removed

File: templates/post.html (hunk 8 above)
Comment (grahambancroft) [reply templates/post.html]: Added

File: templates/post.html (hunk 7 above)
Comment (grahambancroft) [reply templates/post.html]: Fixed

File: templates/post.html [hunk 10]
```diff
@@ -87,6 +99,70 @@ <h3>
...
+            </ul>
+
+          <div class='mktFormReq mktField'>
+            <label for="Email" class="mktoLabel">Work email: <span>*</span></label>
+            <input required  id="Email" name="Email" maxlength="255" type="email" class="mktoField mktoEmailField  mktoRequired" />
+          </div>
+
+          <div class="mktField mktLblRight">
+            <span class="mktInput mktLblRight">
+              <input class="mktFormCheckbox" name="canonicalUpdatesOptIn" id="canonicalUpdatesOptIn" value="Receive updates" tabindex="9" type="checkbox" />
```
Comment (anthonydillon) [inline templates/post.html]: This tabIndex is making it impossible to tab too

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Code looks good :+1: 

Just one QA issue.

File: templates/post.html (hunk 10 above)
Comment (grahambancroft) [reply templates/post.html]: Fixed

**Review by anthonydillon — APPROVED** [review]

**Pull request merged**